### ISSUE TITLE
Hot reload for islands

### ISSUE THREAD

**Issue by m4rocks** [denoland/fresh#316]
Is fresh capable of hot reloading while in dev mode? If not, would it be possible?

**Comment by fulsomenko** [comment]
If by hot reloading you mean the changes appear in your browser when you save a file. Then yes.

**Comment by m4rocks** [comment]
yes i noticed. but I mean could the framework only refresh the component that needs to be updated? this is considered it is used with preact

**Comment by lucacasonato** [comment]
It is not capable of doing this now. It may be possible that we can support this for islands in the future, but Fresh will never support this for generic pages, as there is no infrastructure for rendering pages on the client (pages are always server side rendered).

**Comment by m4rocks** [comment]
well yes, I would not expect hot reloading for prerendered content. but it would be nice for islands

**Comment by JLarky** [comment]
<img width="626" alt="image" src="https://user-images.githubusercontent.com/7026/177023178-58283cd3-f91d-4692-81f9-051dc70347c1.png">

have anyone tried something like that? :)

https://twitter.com/RyanCarniato/status/1542688476654493696

**Comment by zoeleu** [comment]
This would be possible by injecting a WebSocket script, and telling the browser to refresh with every change. This would be pretty bare bones, and doesn't maintain state, but hey, it would be a start.

**Comment by lucacasonato** [comment]
> This would be possible by injecting a WebSocket script, and telling the browser to refresh with every change. This would be pretty bare bones, and doesn't maintain state, but hey, it would be a start.

It already does this :)

**Comment by zoeleu** [comment]
Weird, it's not refreshing when I do changes.

**Comment by m4rocks** [comment]
i'm doing some research on how preact handles hot reloading aka preferesh. i'm working on a fix now and will open a pr soon

**Comment by dector** [comment]
@matteoturini 

> Weird, it's not refreshing when I do changes.

If you are using FF - check #694

**Comment by marcelocra** [comment]
_(Just adding Hot Module Replacement - HMR - here to make it easy for people looking for that in Fresh. Now it will appear in GitHub issue search results.)_

> It already does this :)

@lucacasonato What do you mean by this? Does it maintain island state? Because if so, it is not currently working for me.

**Comment by jiawei397** [comment]
> i'm doing some research on how preact handles hot reloading aka preferesh. i'm working on a fix now and will open a pr soon

Do you have any news now?

**Comment by marvinhagemeister** [comment]
Resolved by https://github.com/denoland/fresh/pull/3187
